qt/RecentDocuments.cpp

```diff
@@ -0,0 +1,155 @@
+/* -*- Mode: C++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4; fill-column: 100 -*- */
+/*
+ * Copyright the Collabora Online contributors.
+ *
+ * SPDX-License-Identifier: MPL-2.0
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include "RecentDocuments.hpp"
+#include "Log.hpp"
+#include "LibreOfficeKit/LibreOfficeKit.h"
+
+#include <QFileInfo>
+#include <QSettings>
+#include <QUrl>
+#include <QDateTime>
+#include <Poco/JSON/Object.h>
+
+namespace
+{
+    constexpr int MAX_RECENT = 15;
+    constexpr const char* SETTINGS_ORG = "Collabora";
+    constexpr const char* SETTINGS_APP = "CODA-RecentDocuments";
+    constexpr const char* KEY_URIS = "uris";
+    constexpr const char* KEY_NAMES = "names";
+    constexpr const char* KEY_TIMESTAMPS = "timestamps";
+}
+int RecentDocuments::getAppTypeFromExtension(const QString& suffix)
+{
+    QString ext = suffix.toLower();
+    if (ext == "odt" || ext == "ott" || ext == "doc" || ext == "docx" ||
+        ext == "rtf" || ext == "txt" || ext == "odm" || ext == "fodt")
+        return LOK_DOCTYPE_TEXT;
+    if (ext == "ods" || ext == "ots" || ext == "xls" || ext == "xlsx" ||
+        ext == "csv" || ext == "fods")
+        return LOK_DOCTYPE_SPREADSHEET;
+    if (ext == "odp" || ext == "otp" || ext == "ppt" || ext == "pptx" ||
+        ext == "fodp")
+        return LOK_DOCTYPE_PRESENTATION;
+    return LOK_DOCTYPE_TEXT; // default fallback
+}
+
+QString RecentDocuments::getGroupName(int docType)
+{
+    switch (docType) {
+        case LOK_DOCTYPE_TEXT:        return "RecentDocuments_Writer";
+        case LOK_DOCTYPE_SPREADSHEET: return "RecentDocuments_Calc";
+        case LOK_DOCTYPE_PRESENTATION: return "RecentDocuments_Impress";
+        default:                      return "RecentDocuments_Writer";
+    }
+}
+
+QString RecentDocuments::getDocTypeString(int docType)
+{
+    switch (docType) {
+        case LOK_DOCTYPE_SPREADSHEET: return "calc";
+        case LOK_DOCTYPE_PRESENTATION: return "impress";
+        case LOK_DOCTYPE_TEXT:
+        default: return "writer";
```

Comment:
looks oddly inconsistent how `getDocTypeString` here combines `csae LOK_DOCTYPE_TEXT` and `default`, while the above `getGroupName` keeps them apart